### PR TITLE
Handle unexpected keywords in expressions

### PR DESCRIPTION
**Pull request by dav000**
#407 

The new behavior is to end the expression for these keywords. It will not report an unexpected token error in an expression. 

Because the code continues parsing instead of crashing, there are known new issues with handling `catch`, `export`, `extends`, `finally` and `function` expressions. These issues occur when keywords are mentioned in unopened comment blocks.

For example:

```
*                  
* this function extends a string 
* /
```

### CODE REVIEW

**Comment by strager** [comment]
How did you test this change?

> Because the code continues parsing instead of crashing, there are known new issues with handling `catch`, `export`, `extends`, `finally` and `function` expressions.

Can you include a test case demonstrating the behavior (and why it might not be desirable)?

**Comment by strager** [comment]
Maybe add some test cases to `incomplete_unary_expression_with_following_statement_keyword`?

**Comment by dav000** [comment]
Attached the test file I used to confirm quick-lint-js doesn't crash. there are a few keywords commented out in the test file because they result in crashes in downstream code specific to the keyword. because those aren't simple fixes, they will need to have independent issues opened.
[kw4.txt](https://github.com/quick-lint/quick-lint-js/files/6948593/kw4.txt)

File: test/test-parse-expression-statement.cpp [hunk 1]
```diff
@@ -1159,6 +1159,37 @@ TEST(test_parse, trailing_comma_in_comma_expression_is_disallowed) {
...
+           u8"break"_sv,
+           u8"case"_sv,
+           u8"const"_sv,
+           u8"continue"_sv,
+           u8"default"_sv,
+           u8"export"_sv,
+           u8"try"_sv,
+           u8"var"_sv,
+       }) {
+    padded_string code(string8(u8"*\n"_sv) + string8(statement));
```
Comment (strager) [inline test/test-parse-expression-statement.cpp:1184]: Nit:
```suggestion
  for (string8 statement : {
           u8"break"_sv,
           u8"case"_sv,
           u8"const"_sv,
           u8"continue"_sv,
           u8"default"_sv,
           u8"export"_sv,
           u8"try"_sv,
           u8"var"_sv,
       }) {
    padded_string code(u8"*\n"_sv + statement);
```

**Review by strager — APPROVED** [review]

File: test/test-parse-expression-statement.cpp (hunk 1 above)
Comment (dav000) [reply test/test-parse-expression-statement.cpp:1184]: the suggested code change doesn't compile

**Pull request merged**